src/main/webapp/app/exercises/shared/structured-grading-criterion/structured-grading-criterion.service.ts

```diff
@@ -32,23 +32,23 @@ export class StructuredGradingCriterionService {
     }
 
     calculateScoreForGradingInstructions(feedback: Feedback, score: number, gradingInstructions: any): number {
-        if (gradingInstructions[feedback.gradingInstruction!.id]) {
+        if (gradingInstructions[feedback.gradingInstruction!.id!]) {
             // We Encountered this grading instruction before
             const maxCount = feedback.gradingInstruction!.usageCount;
-            const encounters = gradingInstructions[feedback.gradingInstruction!.id];
+            const encounters = gradingInstructions[feedback.gradingInstruction!.id!];
             if (maxCount && maxCount > 0) {
                 if (encounters >= maxCount) {
-                    gradingInstructions[feedback.gradingInstruction!.id] = encounters + 1;
+                    gradingInstructions[feedback.gradingInstruction!.id!] = encounters + 1;
                 } else {
-                    gradingInstructions[feedback.gradingInstruction!.id] = encounters + 1;
+                    gradingInstructions[feedback.gradingInstruction!.id!] = encounters + 1;
```

Comment:
Same thing happens inside both if and else statements. I don't know if you are going to do anything about it since you resolved it. I just wanted to point it out again in case you forgot it.